src/pages/index.tsx

```diff
@@ -1,20 +1,31 @@
 import { useState } from "react"
 import MainButton from "@/components/buttons/main-button"
 import ErrorPopup from "@/components/popups/error-popup"
-import WithProtectedRoute from "@/components/protected-route/protected-route"
-import { signOutUser } from "@/services/authentication.service"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { useRouter } from "next/router"
 
 function Home() {
+  const router = useRouter()
   const [showError, setShowError] = useState<boolean>(false)
+  const supabase = useSupabaseClient<Database>()
+
+  function signOut() {
+    ;(async () => {
```

Comment:
;

Reply:
IDE automatically adds it because of this:
https://stackoverflow.com/questions/1873983/what-does-the-leading-semicolon-in-javascript-libraries-do